OneSignalSDK/onesignal/notifications/src/main/java/com/onesignal/notifications/internal/permissions/impl/NotificationPermissionController.kt

```diff
@@ -84,6 +85,10 @@ internal class NotificationPermissionController(
      * to notify of the status.
      */
     override suspend fun prompt(fallbackToSettings: Boolean): Boolean {
+        // Calling yield() to force a suspension point because Kotlin Continuation won't work
```

Comment:
nit, could we point out this is to specifically address calls from Java?
* You're are correct that the issue with Kotlin's Continuation but bring up the Java caller make is very clear at a glance.